CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,184 @@
+
+Approach (2) was deemed risky and complicated. Approach (1) felt more in line with the goal of having Granda Mento be simple and safe, but ultimately the involvement of Governance for every exchange felt unsustainable and exhausting for voters. The proposed implementation involving an exchange delay, a multisig for approvals, and Governance in the (likely) rare event the community wishes to veto an exchange, aims to relieve voter exhaustion while still providing the security of community scrutiny.
+
+The proposed implementation prohibits the proposer from cancelling their own exchange proposal after it has been approved. This is to prevent Granda Mento from being used as a "free option," where the proposer can decide against the exchange late in the process if the price of the exchange is no longer favorable.
+
+[Initial discussions around Granda Mento](https://forum.celo.org/t/discussion-on-granda-mento-enabling-larger-stablecoin-mints/966) showed concern around providing a "fair" price to ensure that large players do not have an unfair opportunity to "dump" CELO for stable tokens. Some ideas for pricing were considered:
+
+1. Oracle price at the time of exchange proposal.
+   * This is vulnerable to oracle attacks.
+2. The price is included by the exchange proposer, and is required to be within X% of the current oracle price.
+   * This is vulnerable to oracle attacks.
+   * This puts more control in the hands of the exchange proposer.
+3. Oracle price at the time of exchange execution.
+   * This is vulnerable to oracle attacks.
+   * Exchanger doesn't know the price at the proposal time.
+4. TWAP between exchange proposal and exchange execution.
+   * Involves smart contract changes to have the TWAP available on chain.
+   * Exchanger doesn't know the price at the proposal time.
+
+It's desirable for both the exchanger and for the Celo community to know what price will be used for the exchange-- this way, the exchanger knows what they're committing to, and the community can decide if they agree with the price. Approaches (1) and (2) the only options that involve knowledge of the price at the start of the trade.
+
+Another question is whether Granda Mento should have an "allowance" that must be granted by Governance indicating how many stable tokens it can mint/burn. Each exchange through Granda Mento would "spend" some of that allowance, and eventually a top-up would be required. This allows reasoning about the worst-case scenario if the approver multisig is compromised and Governance is unable to rally in time. Because this would involve more active involvement from Governance for top-ups and some additional implementation complexity, instead a configurable value in Granda Mento is proposed to restrict how many stable tokens a single exchange can mint/burn.
```

Comment:
I think we can leave this "allowance" out. Imo the max exchange amount largely addresses this concern and an allowance is unnecessary.